include/macros.h

```diff
@@ -21,10 +21,7 @@
 
 #define ARRAY_COUNT_2D(arr) (ARRAY_COUNT(arr) * ARRAY_COUNT(arr[0]))
 
-// TODO: After uintptr_t cast change should have an AVOID_UB target that just toggles the KSEG0 bit in the address rather than add/sub 0x80000000
-#define PHYSICAL_TO_VIRTUAL(addr) ((uintptr_t)(addr) + RDRAM_CACHED)
-#define VIRTUAL_TO_PHYSICAL(addr) (uintptr_t)((u8*)(addr) - RDRAM_CACHED)
-#define SEGMENTED_TO_VIRTUAL(addr) (void*)(PHYSICAL_TO_VIRTUAL(gSegments[SEGMENT_NUMBER(addr)]) + SEGMENT_OFFSET(addr))
+#define SEGMENTED_TO_VIRTUAL(addr) (void*)(((uintptr_t)(gSegments[SEGMENT_NUMBER(addr)]) + K0BASE) + SEGMENT_OFFSET(addr))
```

Comment:
```suggestion
#define SEGMENTED_TO_VIRTUAL(addr) (void*)((gSegments[SEGMENT_NUMBER(addr)] + K0BASE) + SEGMENT_OFFSET(addr))
```
Any particular reason for the `uintptr_t` cast?

Reply:
This seems to be a side-effect of removing the other macros. It isn't required